include/irr/core/math/aabb2dSIMDIntBase.h

```diff
@@ -216,9 +175,80 @@ class aabbSIMD<core::vector2df_SIMD> : public aabbSIMDBase<aabbSIMD, core::vecto
 
 };
 
+template<typename VECTOR_TYPE>
+class aabb2dSIMDInt : public aabbSIMDBase<aabb2dSIMDInt, VECTOR_TYPE>
+{
+	static_assert(
+		std::is_same<VECTOR_TYPE, core::vector2di32_SIMD>::value ||
+		std::is_same<VECTOR_TYPE, core::vector2du32_SIMD>::value,
+		"assertion failed: cannot use this type");
+
+public:
+	inline aabb2dSIMDInt()
```

Comment:
only this function will vary between aabb2dSIMDi and aabb2dSIMDu i believe

Reply:
sneaky way of solving this ;)

Reply:
if you implement all the comparison operators in the vectorSIMD32* types then yes.

Reply:
thats the plan, but implementing those will take some time